datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -729,8 +743,15 @@ def _submit_gauges_from_histogram(self, metric_name, metric, scraper_config, hos
         """
         Extracts metrics from a prometheus histogram and sends them as gauges
         """
+
+        cumulative_metric = None
         if scraper_config['non_cumulative_buckets']:
+            if scraper_config['send_histogram_counters_with_distributions']:
+                # make a cumlative copy of the original metric so that we can send cumlative bucket counters after we
+                # send the distributions
+                cumulative_metric = deepcopy(metric)
             self._decumulate_histogram_buckets(metric)
```

Comment:
I feel like `metric` should be immutable and this should return a new metric altogether rather than modify the existing one. Took me awhile to understand why my counters weren't working because I overlooked this "side effect".